File: Example/src/main/java/com/app/herysapps/calculatordialog/ExampleActivity.java
package com.app.herysapps.calculatordialog;

import android.os.Bundle;

import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.app.herysapps.calculatordialoglib.CalculatorDialog;
import com.app.herysapps.calculatordialoglib.CalculatorDialogBuilder;

/**
 * Examples of use.
 * <p>
 * Created by Hery Lopez on 10/08/2017.
 * <p>
 * Github:
 * Author: https://github.com/HeryLopez
 * Project: https://github.com/HeryLopez/CalculatorDialog
 */
public class ExampleActivity extends AppCompatActivity implements CalculatorDialog.OnDialogResultListener {

    CalculatorDialog cal01, cal02;

    private String DIALOG_01 = "calculatorDialog01";
    private String DIALOG_02 = "calculatorDialog02";

    private double v1, v2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_example);

        CalculatorDialogBuilder builderDialog1 = new CalculatorDialogBuilder();
        builderDialog1.setOnResultListener(this);
        builderDialog1.limitNumbers(20);
        builderDialog1.negativeNumberActivated(true);
        builderDialog1.setErrorDiv0(getString(R.string.div_0_error));
        builderDialog1.setErrorLimit(getString(R.string.limit_number_error));
        builderDialog1.setErrorNegativeValue(getString(R.string.limit_negative_number));

        cal01 = builderDialog1.build();

        v1 = 1234567.89; //3.1415
        String strWithFormat = cal01.getNumberWithFormat(v1);
        ((TextView)findViewById(R.id.textView1)).setText(strWithFormat);


        (findViewById(R.id.buttonSelector01)).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                cal01.showDialog(getSupportFragmentManager(), DIALOG_01, v1);
            }
        });

        CalculatorDialogBuilder builderDialog2 = new CalculatorDialogBuilder();
        builderDialog2.setOnResultListener(this);
        builderDialog2.setDecor("$");
        builderDialog2.setNumberColor(R.color.color01);
        builderDialog2.setOperationColor(R.color.color01);
        builderDialog2.setNumberBackgroundColor(R.color.color02);
        builderDialog2.setOperatorBackgroundColor(R.color.color02);
        builderDialog2.setDialogButtonsColor(R.color.color01);

        cal02 = builderDialog2.build();

        (findViewById(R.id.buttonSelector02)).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                cal02.showDialog(getSupportFragmentManager(), DIALOG_02, v2);
            }
        });
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
    }

    @Override
    public void onDialogResult(String tagDialog, double value, String valueStr) {

        if (tagDialog.equals(DIALOG_01)) {
            ((TextView)findViewById(R.id.textView1)).setText(valueStr);
            v1 = value;
        }

        if (tagDialog.equals(DIALOG_02)) {
            ((TextView)findViewById(R.id.textView2)).setText(valueStr);
            v2 = value;
        }
    }
}

File: Example/src/main/java/com/app/herysapps/calculatordialog/ExampleActivity.kt
package com.app.herysapps.calculatordialog

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentTransaction
import com.app.herysapps.calculatordialoglib.CalculatorDialog
import com.app.herysapps.calculatordialoglib.CalculatorDialog.OnDialogResultListener
import com.app.herysapps.calculatordialoglib.CalculatorDialogBuilder
import kotlinx.android.synthetic.main.activity_example.*

/**
 * Created by Hery Lopez on 10/08/2017.
 *
 * Github:
 * Author: https://github.com/HeryLopez
 * Project: https://github.com/HeryLopez/CalculatorDialog
 */
class ExampleActivity : AppCompatActivity(), OnDialogResultListener {

    var dialogA: CalculatorDialog? = null
    var dialogB: CalculatorDialog? = null
    private val ID_DIALOG_EXAMPLE_1 = "calculatorDialog01"
    private val ID_DIALOG_EXAMPLE_2 = "calculatorDialog02"
    private val ID_DIALOG_EXAMPLE_3 = "calculatorDialog03"
    private var v1 = 0.0
    private var v2 = 0.0
    private var v3 = 0.0

    private var isFullscreenModelActive = false

    private fun buildDialog() {
        val builderDialog1 = CalculatorDialogBuilder()
        builderDialog1.setOnResultListener(this)
        builderDialog1.limitNumbers(20)
        builderDialog1.negativeNumberActivated(true)
        builderDialog1.setErrorDiv0("Division par 0 impossible")
        builderDialog1.setErrorLimit("Limite de nombre dépassée")
        builderDialog1.setErrorNegativeValue("Numéros négatifs désactivés")
        builderDialog1.setOkBackgroundColor(R.color.colorPrimaryDark)
        dialogA = builderDialog1.build()


        val builderDialog2 = CalculatorDialogBuilder()
        builderDialog2.setOnResultListener(this)
        builderDialog2.setDecor("$")
        builderDialog2.setNumberColor(R.color.color01)
        builderDialog2.setOperationColor(R.color.color01)
        builderDialog2.setNumberBackgroundColor(R.color.color02)
        builderDialog2.setOperatorBackgroundColor(R.color.color02)
        builderDialog2.setOkBackgroundColor(R.color.colorAccent)
        dialogB = builderDialog2.build()
    }

    private fun openFullscreenDialog(dialog: CalculatorDialog, name: String) {
        val transaction = supportFragmentManager.beginTransaction()
        transaction.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN)
        transaction.replace(android.R.id.content, dialog).addToBackStack(null).commit()
    }

    private fun example1() {
        v1 = 1234567.89 //3.1415
        val strWithFormat = dialogA!!.getNumberWithFormat(v1)
        textView1.text = strWithFormat
        buttonSelector01.setOnClickListener {
            dialogA!!.setInitialValue(v1)
            dialogA!!.setDialogId(ID_DIALOG_EXAMPLE_1)
            if (isFullscreenModelActive) {
                openFullscreenDialog(dialogA!!, ID_DIALOG_EXAMPLE_1)
            } else {
                dialogA!!.show(supportFragmentManager, ID_DIALOG_EXAMPLE_1)
            }
        }
    }

    private fun example2() {
        buttonSelector02.setOnClickListener {
            dialogA!!.setInitialValue(v2)
            dialogA!!.setDialogId(ID_DIALOG_EXAMPLE_2)
            if (isFullscreenModelActive) {
                openFullscreenDialog(dialogA!!, ID_DIALOG_EXAMPLE_2)
            } else {
                dialogA!!.show(supportFragmentManager, ID_DIALOG_EXAMPLE_2)
            }
        }
    }

    private fun example3() {
        v3 = 0.0 //3.1415
        val strWithFormat = dialogA!!.getNumberWithFormat(v1)
        textView3.text = strWithFormat
        buttonSelector03.setOnClickListener {
            dialogB!!.setInitialValue(v3)
            dialogB!!.setDialogId(ID_DIALOG_EXAMPLE_3)
            if (isFullscreenModelActive) {
                openFullscreenDialog(dialogB!!, ID_DIALOG_EXAMPLE_3)
            } else {
                dialogB!!.show(supportFragmentManager, ID_DIALOG_EXAMPLE_3)
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_example)

        buildDialog()

        switchFullscreen.setOnCheckedChangeListener { _, isChecked ->
            isFullscreenModelActive = isChecked
        }

        example1()
        example2()
        example3()
    }


    override fun onDialogResult(tagDialog: String, value: Double, valueStr: String) {
        if (tagDialog == ID_DIALOG_EXAMPLE_1) {
            textView1.text = valueStr
            v1 = value
        }
        if (tagDialog == ID_DIALOG_EXAMPLE_2) {
            textView2.text = valueStr
            v2 = value
        }
        if (tagDialog == ID_DIALOG_EXAMPLE_3) {
            textView3.text = valueStr
            v3 = value
        }
    }
}